identity-service/src/routes/tiktok.js

```diff
@@ -57,9 +64,159 @@ module.exports = function (app) {
+
+      try {
+        const tikTokObj = await models.TikTokUser.findOne({
+          where: { uuid: uuid }
+        })
+
+        const user = await models.User.findOne({
+          where: { handle }
+        })
+
+        const isUnassociated = tikTokObj && !tikTokObj.blockchainUserId
+        const handlesMatch =
+          tikTokObj &&
+          tikTokObj.profile.display_name.toLowerCase() ===
+            user.handle.toLowerCase()
+
+        // only set blockchainUserId if not already set
+        if (isUnassociated && handlesMatch) {
+          tikTokObj.blockchainUserId = userId
+
+          // if the user is verified, write to chain, otherwise skip to next step
+          if (tikTokObj.verified) {
+            const [encodedABI, contractAddress] =
+              await audiusLibsInstance.User.updateIsVerified(
+                userId,
+                true,
+                config.get('userVerifierPrivateKey'),
+                config.get('entityManagerReplicaSetEnabled')
+              )
+            const contractRegKey =
+              await audiusLibsInstance.contracts.getRegistryContractForAddress(
+                contractAddress
+              )
+            const senderAddress = config.get('userVerifierPublicKey')
+
+            try {
+              const txProps = {
+                contractRegistryKey: contractRegKey,
+                contractAddress: contractAddress,
+                encodedABI: encodedABI,
+                senderAddress: senderAddress,
+                gasLimit: null
+              }
+              await txRelay.sendTransaction(
+                req,
+                false,
+                txProps,
+                'tikTokVerified'
+              )
+              await verifiedUserReporter.report({ userId, handle })
+            } catch (e) {
+              return errorResponseBadRequest(e)
+            }
+          }
+
+          const socialHandle = await models.SocialHandles.findOne({
+            where: { handle }
+          })
+          if (socialHandle) {
+            socialHandle.tikTokHandle = tikTokObj.profile.display_name
+            await socialHandle.save()
+          } else if (tikTokObj.profile && tikTokObj.profile.display_name) {
+            await models.SocialHandles.create({
+              handle,
+              tikTokHandle: tikTokObj.profile.display_name
+            })
+          }
+
+          // the final step is to save userId to db and respond to request
+          try {
+            await tikTokObj.save()
+            return successResponse()
+          } catch (e) {
+            return errorResponseBadRequest(e)
+          }
+        } else {
+          req.logger.error(
+            `TikTok profile does not exist or userId has already been set for uuid: ${uuid}`,
+            tikTokObj
+          )
+          return errorResponseBadRequest(
+            'TikTok profile does not exist or userId has already been set'
+          )
+        }
+      } catch (err) {
```

Comment:
Awesome job with this route!

Reply:
Thank you :) mostly copied from twitter/ig